code/game/objects/effects/decals/decal.dm

```diff
@@ -45,7 +45,17 @@
 	loc.AddElement(/datum/element/decal, icon, icon_state, dir, FALSE, color, TURF_LAYER + (layer - TURF_DECAL_LOWEST_LAYER), null, alpha, FALSE)
 	return INITIALIZE_HINT_QDEL
 
-/obj/effect/turf_decal/Destroy()
+/obj/effect/turf_decal/Destroy(force)
 	SHOULD_CALL_PARENT(FALSE)
+#ifdef REFERENCE_TRACKING_FAST
+// If we don't do this, turf decals will end up stacking up on a tile, and break the overlay limit
+// I hate it too bestie
+	if(GLOB.running_create_and_destroy)
+		var/turf/T = loc
+		T.RemoveElement(/datum/element/decal, icon, icon_state, dir, null, layer, alpha, color, null, FALSE, null)
+#endif
```

Comment:
This is pretty horrendous. Can you make it just never apply decals instead of UNIT_TESTS is defined and remove the GLOB.running_create_and_destroy var?

Reply:
I doubt if it's a good idea since it's 100% TG port.
This global variable is specifically for when `/datum/unit_test/create_and_destroy` is running but also when decals are being created. When `/datum/unit_test/create_and_destroy` test is over, we can possibly have more decals in creation.